Lib/glyphsLib/classes.py

```diff
@@ -3227,46 +3283,47 @@ def exports(self):
     def exports(self, value):
         self.active = value
 
-    def _get_from_properties(self, key):
-        for p in self.properties:
-            if p.key == key:
-                return p.defaultValue
-        return None
-
     @property
     def familyName(self):
-        value = (
-            self._get_from_properties("familyNames")
+        return (
+            self.properties.get("familyNames")
             or self.customParameters["familyName"]
+            or self.parent.familyName
         )
-        if value:
-            return value
-        return self.parent.familyName
 
     @familyName.setter
     def familyName(self, value):
+        # TODO: Update this to write into either custom parameters or properties
+        # depending on the version.
         self.customParameters["familyName"] = value
 
     @property
     def preferredFamily(self):
-        value = self.customParameters["preferredFamily"]
-        if value:
-            return value
-        return self.parent.familyName
+        return self.customParameters["preferredFamily"] or self.parent.familyName
```

Comment:
from my experiments, I believe this should be like below, because it takes into account the properties if present, otherwise falls back to the default familyName:

```suggestion
        return self.preferredFamilyName or self.parent.familyName
```

Reply:
Changed.